tests/MMS/spatial/advection/runtest

```diff
@@ -13,180 +13,172 @@ from builtins import str
+    # sizes, which is suspicious. It does seem to work correctly in
+    # other contexts, so not overly worrying that it doesn't work
+    # exactly as expected here
+    # ("method=1 mesh:ddx:upwind=w3 mesh:ddz:upwind=w3", "3rd order WENO", "-s", 3),
+]
 
 # List of NX values to use
-nxlist = [16, 32]
-#nxlist = [16, 32, 64, 128, 256, 512, 1024]
+nxlist = [16, 32, 64, 128, 256]
 
 nproc = 2
+mthread = 2
 
 success = True
 
 err_2_all = []
 err_inf_all = []
-for opts,label,sym,exp_ord in options:
-    error_2   = []  # The L2 error (RMS)
+for opts, label, sym, exp_ord in options:
+    error_2 = []  # The L2 error (RMS)
     error_inf = []  # The maximum error
 
     for nx in nxlist:
-        # Set the X and Z mesh size
+        shell("rm data/BOUT.dmp.*.nc")
 
+        # Set the X and Z mesh size
         dx = 2.*pi / (nx)
-
         args = opts + " mesh:nx="+str(nx+4)+" mesh:dx="+str(dx)+" MZ="+str(nx)
-        
         print("  Running with " + args)
-
-        # Delete old data
-        shell("rm data/BOUT.dmp.*.nc")
-        
-        # Command to run
         cmd = "./advection "+args
-        # Launch using MPI
-        s, out = launch_safe(cmd, nproc=nproc, pipe=True)
-
-        # Save output to log file
-        f = open("run.log."+str(nx), "w")
-        f.write(out)
-        f.close()
-        
-        # Collect data
-        E_f = collect("E_f", tind=[1,1], info=False, path="data")
-        #E_f = E_f[0,2:-2,0,:]
-        
-        # Average error over domain
-        l2 = sqrt(mean(E_f**2))
-        linf = max(abs( E_f ))
-        
-        error_2.append( l2 )
-        error_inf.append( linf )
-        
+        s, out = launch_safe(cmd, nproc=nproc, mthread=mthread, pipe=True)
+
+        with open("run.log."+str(nx), "w") as f:
+            f.write(out)
+
+        l2 = collect("l_2", tind=-1, xguards=False,
+                     yguards=False, info=False, path="data")
+        linf = collect("l_inf", tind=-1, xguards=False,
+                       yguards=False, info=False, path="data")
+
+        error_2.append(l2)
+        error_inf.append(linf)
+
         print("  -> Error norm: l-2 %f l-inf %f" % (l2, linf))
-        
-    
-    # Append to list of all results
-    err_2_all.append( (error_2, label, sym) )
-    err_inf_all.append( (error_inf, label, sym) )
 
-    # Calculate grid spacing
+    err_2_all.append((error_2, label, sym))
+    err_inf_all.append((error_inf, label, sym))
+
     dx = 1. / array(nxlist)
 
-    # Calculate convergence order
-    
-    order = log(error_2[-1] / error_2[-2]) / log(dx[-1] / dx[-2])
-    print("Convergence order = %f" % (order))
+    order = check_scaling.get_order(dx, error_2)
+    print("Convergence order = {:f} ({:f} at small spacing)".format(*order))
 
-    if not ( exp_ord + .2 > order > exp_ord - .2):
-        success=False
-        print("... Failure")
-    else:
+    success = check_scaling.check_order(order[0], exp_ord)
```

Comment:
note that this will overwrite the previous success state, thus potentially leading to a clean exit, when it shouldn't.
```
if check_scaling...
    print("... Success")
else:
   success = False
   print("... Failure")
```
should fix it